protocol/messenger_handler.go

```diff
@@ -1523,6 +1523,19 @@ func (m *Messenger) HandleChatMessage(state *ReceivedMessageState) error {
 	if err != nil {
 		return fmt.Errorf("failed to prepare message content: %v", err)
 	}
+
+	// If the user was not mentionned, we still check if there is a reply as they count as mentions as well
+	if !receivedMessage.Mentioned && receivedMessage.ResponseTo != "" {
+		repliedTo, err := m.persistence.MessageByID(receivedMessage.ResponseTo)
+		if err != nil {
+			logger.Warn("failed to get quoted message", zap.Error(err))
```

Comment:
I think we can fail fast here and return `err`

Reply:
Yeah, I wasn't sure if I should. That error only causes the message to not have the `Mentioned` set correctly to `true`, so I didn't want to kill the whole message handling for that. Let me know what you think

Reply:
Figured that! I'm good with this as well

Reply:
we should fail fast on any error that's not expected.
If `MessageByID` returns some known errors (i.e `sql.ErrNoRows` because for example the message are arriving out of order), we should handle it explicitly, but if the error is unexpected, better to return the error and stop processing.